core/src/main/scala/org/apache/spark/scheduler/OutputCommitCoordinator.scala

```diff
@@ -111,13 +115,13 @@ private[spark] class OutputCommitCoordinator(conf: SparkConf, isDriver: Boolean)
     val arr = new Array[TaskAttemptNumber](maxPartitionId + 1)
     java.util.Arrays.fill(arr, NO_AUTHORIZED_COMMITTER)
     synchronized {
-      authorizedCommittersByStage(stage) = arr
+      stageStates(stage) = new StageState(arr)
```

Comment:
ah sorry now that I see this I realized it probably makes sense to initialize `arr` in the StageState constructor too (so this line would look like `new StageState(maxPartitionId +1)`, and the `StageState` constructor just takes in `numPartitions`).  Would you mind making that change too?